model_api/cpp/CMakeLists.txt

```diff
@@ -2,6 +2,74 @@
 # SPDX-License-Identifier: Apache-2.0
 #
 
-add_subdirectory(utils)
-add_subdirectory(models)
-add_subdirectory(pipelines)
+cmake_minimum_required(VERSION 3.10)
+
+# Multi config generators such as Visual Studio ignore CMAKE_BUILD_TYPE. Multi config generators are configured with
+# CMAKE_CONFIGURATION_TYPES, but limiting options in it completely removes such build options
+get_property(GENERATOR_IS_MULTI_CONFIG_VAR GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG)
+if(NOT GENERATOR_IS_MULTI_CONFIG_VAR AND NOT DEFINED CMAKE_BUILD_TYPE)
+    message(STATUS "CMAKE_BUILD_TYPE not defined, 'Release' will be used")
+    # Setting CMAKE_BUILD_TYPE as CACHE must go before project(). Otherwise project() sets its value and set() doesn't take an effect
+    set(CMAKE_BUILD_TYPE Release CACHE STRING "Choose the type of build, options are: None Debug Release RelWithDebInfo MinSizeRel ...")
+endif()
+
+project(model_api)
+
+if(WIN32)
+    if(NOT "${CMAKE_SIZEOF_VOID_P}" EQUAL "8")
+        message(FATAL_ERROR "Only 64-bit supported on Windows")
+    endif()
+
+    add_definitions(-DNOMINMAX)
+endif()
+
+set(model_api_VERSION 0.0.0)
+
+find_package(OpenCV REQUIRED COMPONENTS core imgproc)
+find_package(OpenVINO REQUIRED COMPONENTS Runtime)
```

Comment:
these `COMPONENTS` are not synchronized with `find_package` calls ins `model_api/cpp/cmake/model_apiConfig.cmake`